mitosheet/mitosheet/public/v3/sheet_functions/bool_functions.py

```diff
@@ -123,6 +123,52 @@ def IF(condition: pd.Series, true_series: AnyPrimitiveOrSeriesInputType, false_s
     )
 
 
+@cast_values_in_arg_to_type('argv', 'bool')
+@handle_sheet_function_errors
+def IFS(*argv: Optional[IfsInputType]) -> pd.Series:
+    """
+    {
+        "function": "IFS",
+        "description": "Returns one value if the condition is True. Returns the other value if the conditon is False.",
+        "search_terms": ["ifs", "if", "conditional", "and", "or"],
+        "examples": [
+            "IFS(Status == 'success', 1, 0)",
+            "IFS(Nums > 100, 100, Nums)",
+            "IFS(AND(Grade >= .6, Status == 'active'), 'pass', 'fail')"
+        ],
+        "syntax": "IFS(boolean_condition_1, value_if_true, boolean_condition_2, value_if_true)",
+        "syntax_elements": [{
+                "element": "boolean_condition",
+                "description": "An expression or series that returns True or False values. Valid conditions for comparison include ==, !=, >, <, >=, <=."
+            },
+            {
+                "element": "value_if_true",
+                "description": "The value the function returns if condition is True."
+            }
+        ]
+    }
+    """
+    # Go through all of the conditions. The first condition that is true for
+    # *each cell* should be selected. 
+    results = pd.Series()
+    for index, condition in enumerate(argv):
+        # The args are alternating between the conditions and the values corresponding to them.
+        # If this is a series, we assume it is the condition. We skip the values because they're
+        #   only used as a function of the conditions. 
+        if isinstance(condition, pd.Series):
+            # For each cell, check if the value is true.
+            # If it is, use the "true_series" to fill the value in the end series.
+            # If it isn't save this for the next series by using "None"
+            true_series = get_series_from_primitive_or_series(argv[index+1], condition.index)
+            # Add the new data into the series
+            new_series = pd.Series(
+                data=[true_series.loc[i] if c else None for i, c in condition.items()],
+                index=condition.index
```

Comment:
Yeah, this does look really slow. A cool thing that might work -- using the boolean series as indexers, like this: 

<img width="517" alt="Screen Shot 2023-08-20 at 9 43 30 AM" src="https://github.com/mito-ds/mito/assets/21002431/37101e32-ec58-46a5-8c8f-4dca73059cfc">

`s1` is the condition, and because we just use it to select only some of the `true_series`, we can then just put this series subset (with only some of the index labels defined) into the combine_first function.

I think this will be much faster!

```

Reply:
Ah ok didn't know that syntax! That'll be really helpful